dockstore-common/src/main/java/io/dockstore/common/yaml/constraints/EntryName.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dockstore.common.yaml.constraints;
+
+import io.dockstore.common.ValidationConstants;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+/**
+ * Defines the `EntryName` constraint annotation.
+ */
+@Pattern(regexp = ValidationConstants.ENTRY_NAME_REGEX, message = ValidationConstants.ENTRY_NAME_REGEX_MESSAGE)
+@Size(min = ValidationConstants.ENTRY_NAME_LENGTH_MIN, max = ValidationConstants.ENTRY_NAME_LENGTH_MAX, message = ValidationConstants.ENTRY_NAME_LENGTH_MESSAGE)
+@Target({ ElementType.METHOD, ElementType.FIELD, ElementType.ANNOTATION_TYPE, ElementType.CONSTRUCTOR, ElementType.PARAMETER, ElementType.TYPE_USE })
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = {})
+public @interface EntryName {
+
+    String message () default "";
```

Comment:
would like Javadoc for these for understanding

Reply:
The `message` property, or more detail in the `EntryName` javadoc?

Reply:
I think the variables in general in the various annotation classes. Since they don't extend anything, the meaning doesn't seem obvious to me.

Reply:
Added a link to an overview of custom constrain annotations to the package info file.